packages/cli/src/commands/election/activate.ts

```diff
@@ -30,31 +41,84 @@ export default class ElectionActivate extends BaseCommand {
     await newCheckBuilder(this, forAccount).isSignerOrAccount().runChecks()
 
-    const accounts = await kit.contracts.getAccounts()
-    const signerAccount = await accounts.voteSignerToAccount(forAccount)
+    const signerAccount = await signerToAccount(client, forAccount)
+    const groupsWithPendingVotes = await getGroupsWithPendingVotes(clients, signerAccount)
 
-    const election = await kit.contracts.getElection()
-    const hasPendingVotes = await election.hasPendingVotes(signerAccount)
-    if (hasPendingVotes) {
+    if (groupsWithPendingVotes.length > 0) {
+      let txHashes: Hex[]
       if (res.flags.wait) {
+        const contractCalls = await getAllPendingVotesParameters(
+          { public: client, wallet },
+          groupsWithPendingVotes,
+          signerAccount
+        )
+
+        const epochManager = await getEpochManagerContract(clients)
+        const [currentEpochNumber, [_startBlock, _endBlock, startTimestamp], epochDuration] =
+          await Promise.all([
+            epochManager.read.getCurrentEpochNumber(),
+            epochManager.read.getCurrentEpoch(),
+            epochManager.read.epochDuration(),
+          ])
+
+        const rawTxs: CeloTransactionSerialized[] = []
+        for (const contractCall of contractCalls) {
+          rawTxs.push(
+            await presignWriteContract(wallet, {
+              ...contractCall,
+              nonce: await nonceManager.consume({
+                chainId: client.chain.id,
+                client,
+                address: signerAccount,
+              }),
+            })
+          )
+        }
+
         // Spin until pending votes become activatable.
-        ux.action.start(`Waiting until pending votes can be activated`)
-        while (!(await election.hasActivatablePendingVotes(signerAccount))) {
-          await sleep(1000)
+        const endTimestamp = startTimestamp + epochDuration
+        // NOTE: this is only useful for tests as the date is in the past
+        // `remainingMs` should otherwise never be negative
+        const remainingMs = Math.max(Number(endTimestamp * 1000n) - Date.now(), 0)
+        ux.action.start(
+          'Waiting until pending votes can be activated.' +
+            '\n' +
+            'DO NOT SUBMIT FURTHER TRANSACTIONS FROM THIS WALLET until this command has completed.' +
+            '\n' +
+            `Time until next epoch: ~${humanizeDuration(remainingMs, {})}…`
+        )
+        await sleep(remainingMs)
+        // It takes an epoch for pending votes to be activatable
+        while (currentEpochNumber === (await epochManager.read.getCurrentEpochNumber())) {
+          await sleep(1_000)
         }
         ux.action.stop()
+
+        // NOTE: these may fail if the wallet submit txs meanwhile (eg: nonce would be too low)
+        txHashes = []
+        for (const rawTx of rawTxs) {
+          txHashes.push(await wallet.sendRawTransaction({ serializedTransaction: rawTx }))
+        }
```

Comment:
maybe while this is happening have another `ux.action.start("signing transctions." + flags.useLedger ? "Open your Ledger" : "")`